src/Dapr.PluggableComponents/Components/StateStore/StateStoreStateOptions.cs

```diff
@@ -78,14 +79,36 @@ public sealed record StateStoreStateOptions
     /// </remarks>
     public StateStoreConsistency Consistency { get; init; } = StateStoreConsistency.Unspecified;
 
-    internal static StateStoreStateOptions? FromStateOptions(StateOptions options)
+    internal static StateStoreStateOptions? FromStateOptions(StateOptions? options)
     {
         return options != null
             ? new StateStoreStateOptions
             {
-                Concurrency = (StateStoreConcurrency)options.Concurrency,
-                Consistency = (StateStoreConsistency)options.Consistency
+                Concurrency = FromConcurrency(options.Concurrency),
+                Consistency = FromConsistency(options.Consistency)
             }
             : null;
     }
+
+    internal static StateStoreConcurrency FromConcurrency(StateOptions.Types.StateConcurrency concurrency)
+    {
+        return concurrency switch
+        {
+            StateOptions.Types.StateConcurrency.ConcurrencyFirstWrite => StateStoreConcurrency.FirstWrite,
+            StateOptions.Types.StateConcurrency.ConcurrencyLastWrite => StateStoreConcurrency.LastWrite,
+            StateOptions.Types.StateConcurrency.ConcurrencyUnspecified => StateStoreConcurrency.Unspecified,
+            _ => throw new ArgumentOutOfRangeException(nameof(concurrency), String.Format(CultureInfo.CurrentCulture, "The concurrency \"{0}\" was not recognized.", concurrency))
+        };
```

Comment:
Does it make sense to localize exception messages? That is what this `CultureInfo.CurrentCulture` does, right?

Reply:
The `CultureInfo.CurrentCulture` ensures that any formatting of (non-string) arguments is done using the local culture.  We *might* localize these sort of user-facing strings in the future, so agree that we should at least move them to a resource file to make that process easier.  I filed that as #23.